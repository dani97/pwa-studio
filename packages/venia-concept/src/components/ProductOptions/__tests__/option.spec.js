import React from 'react';
import testRenderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Option from '../option';

jest.mock('../../../classify');
jest.mock('../../../util/getRandomColor');
jest.mock('uuid/v4', () => () => '00000000-0000-0000-0000-000000000000');

const defaultProps = {
    attribute_id: '1',
    attribute_code: 'fashion_color',
    label: 'Color',
    values: [
        {
            label: 'red',
            value_index: 0
        },
        {
            label: 'blue',
            value_index: 1
        }
    ]
};

test('renders Option component correctly', () => {
    const component = testRenderer.create(<Option {...defaultProps} />);
    expect(component.toJSON()).toMatchSnapshot();
});

test('renders a SwatchList if attribute_code prop is "fashion_color"', () => {
    const wrapper = mount(<Option {...defaultProps} />);
    expect(wrapper.find('SwatchList').exists()).toBeTruthy;
    wrapper.unmount();
});

test('renders a TileList if attribute_code prop is not "fashion_color"', () => {
    const props = {
        ...defaultProps,
        attribute_code: 'not_fashion_color'
    };
    const wrapper = mount(<Option {...props} />);

    expect(wrapper.find('TileList').exists()).toBeTruthy();
    wrapper.unmount();
});
