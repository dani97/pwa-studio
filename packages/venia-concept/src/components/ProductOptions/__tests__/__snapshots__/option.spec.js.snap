// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Option component correctly 1`] = `
<div
  className="root"
>
  <h3
    className="title"
  >
    <span>
      Color
    </span>
  </h3>
  <div
    className="root"
  >
    <button
      className="root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "--venia-swatch-bg": "123,123,123",
        }
      }
    />
    <button
      className="root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "--venia-swatch-bg": "123,123,123",
        }
      }
    />
  </div>
  <p
    className="selection"
  >
    
  </p>
</div>
`;
